_posts/2016-05-31-unit-vs-integration-akka-testing.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: post
+name: Unit-vs-Integration-Akka-Testing
+title: 'Unit vs Integration Akka Testing'
+date: 2016-05-31 00:20:00 +00:00
+author: Felipe Fernández
+image:
+    src: /assets/img/custom/blog/time.jpg
+    attribution:
+      text: York railway station by Daida Medina
+      href: https://www.instagram.com/p/nBD197Ho2F/?taken-by=thelastfreeusername
+tags:
+- testing
+- unit
+- integration
+- akka
+
+---
+
+This is the sixth post in the series about integrating sync clients with async systems ([1, ](http://codurance.com/2016/04/28/async-systems-with-sync-clients/)[2, ](http://codurance.com/2016/04/30/akka-basics/)[3, ](http://codurance.com/2016/05/10/finite-state-machines-with-akka/)[4, ](http://codurance.com/2016/05/16/publish-subscribe-model-in-kafka/)[5](http://codurance.com/2016/05/16/coordination-in-akka/)). Here we'll see how to test [Akka actors](http://doc.akka.io/docs/akka/current/scala/actors.html) with different testing styles.
+
+
+## Unit vs Integration Testing
+
+Nowadays everybody agrees with the famous [testing pyramid](http://googletesting.blogspot.com.es/2015/04/just-say-no-to-more-end-to-end-tests.html):
+
+<img src="/assets/img/custom/blog/pyramid.png" alt="Testing pyramid" title="Testing pyramid" class="img img-center img-responsive style-screengrab">
+
+Harder to achieve is an agreement on what integration, unit, functional or acceptance means. It is reasonable as applications are structured differently depending on language, architecture and domain. I'll try to distill some essences though:
+
+* Unit: the key in here is isolation. Some people talks about isolated production code, e.g. a single function or a bunch of methods within a class (one public and the rest private). [Some other people](https://vimeo.com/68375232) talks about isolated tests, i.e. tests that may be executed on parallel as they don't access to any shared resource. A third view is that a unit test execution should be synchronous without any concurrency concern. That's [Akka's view](http://doc.akka.io/docs/akka/current/scala/testing.html):
```

Comment:
I like the term 'isolated' but the word 'independent' in this context might be a more appropriate to use. Because tests should be independent - to be able to run them in parallel is the litmus test.